man/fido2-crypttab.sh

```diff
@@ -8,13 +8,17 @@ sudo systemd-cryptenroll --fido2-device=auto /dev/sdXn
 sudo /usr/lib/systemd/systemd-cryptsetup attach mytest /dev/sdXn - fido2-device=auto
 
 # If that worked, let's now add the same line persistently to /etc/crypttab,
-# for the future.
-sudo bash -c 'echo "mytest /dev/sdXn - fido2-device=auto" >>/etc/crypttab'
+# for the future. We don't want to use the (unstable) /dev/sdX name, so let's
+# figure out a stable link:
+udevadm info --query=property --property=DEVLINKS /dev/sdXn
```

Comment:
use:

udevadm info -r -q symlink  /dev/nvme0n1

but it's weird only doing this after executing two commands referencing the device via /dev/sdXn already...

but frankly this just is a distraction here I think. Let's drop the reference to udevadm here, and just always say `/dev/disk/by-uuid/…`. I think that should be enough.

Reply:
I think it's much more realistic to use `/dev/sdX` (or `/dev/nvme…`, /dev/vdX`, etc.) for local interactive setup. Generally the procedure is that the user does `lsblk` or just knows they have some device type, so they start there. For commands like `cfdisk` or `mkfs.*` you would generally use that, in particular because the UUIDs might not be assigned yet.   The substitution to UUIDs is only done for writing configuration that must remain valid after a reboot.